### DISCUSSION TITLE
How do I listen to onFocusWithin event for Group component?

### DISCUSSION THREAD

**Discussion by axelthat** [adobe/react-spectrum#7249] [Q&A]
```
import { Group } from 'react-aria-components'
<Group onFocusWithin={...} />
```

`onFocusWithin` event isn't available. Is there an alternative way?

**Comment by reidbarber** [comment]
You could use [useFocusWithin](https://react-spectrum.adobe.com/react-aria/useFocusWithin.html) on it.

**Reply by axelthat** [reply]
@reidbarber Thank you for your reply. I did try doing that but I get a lot of typescript errors.
```
const MyGroup: FC<GroupProps> = ({...props}) => {
    const { focusWithinProps } = useFocusWithin({})
    return  <Group {...props} {...focusWithinProps} />
}
```

<img width="1253" alt="image" src="https://github.com/user-attachments/assets/ad2efd1d-de14-4884-a4a1-f1423e3280a9">

**Reply by axelthat** [reply]
Looks like the errors are resolved when the `role` attribute is omitted from `focusWithinProps`.

```
const {
    focusWithinProps: { role, ...focusWithinProps }
} = useFocusWithin({
    onFocusWithinChange: (t) => console.log('Focus Within', t)
})
```
